File: Leetcode/PY/Leetcode807-保持城市天际线.py
class Solution:
    def maxIncreaseKeepingSkyline(self, grid: List[List[int]]) -> int:
        n,m = len(grid),len(grid[0])
        row,col = [0] * n,[0] * m
        for i in range(n):
            for j in range(m):
                row[i] = max(row[i],grid[i][j])
                col[j] = max(col[j],grid[i][j])
        res = 0
        for i in range(n):
            for j in range(m):
                res += min(col[j],row[i]) - grid[i][j]
        return res

File: Leetcode/Rust/Leetcode807-保持城市天际线.rs
impl Solution {
    pub fn max_increase_keeping_skyline(grid: Vec<Vec<i32>>) -> i32 {
        let (mut n,mut m) = (grid.len(),grid[0].len());
        let (mut row,mut col) = (vec![0;n],vec![0;m]);
        for i in 0..n {
            for j in 0..m{
                row[i] = row[i].max(grid[i][j]);
                col[j] = col[j].max(grid[i][j]);
            }
        }
        let mut res = 0;
        for i in 0..n {
            for j in 0..m {
                res += row[i].min(col[j]) - grid[i][j];
            }
        }
        res
    }
}